### DISCUSSION TITLE
Issues creating own map style

### DISCUSSION THREAD

**Discussion by hhornburg** [osmandapp/OsmAnd#20633] [Maps Related]
I am creating my own map style (see https://github.com/hhornburg/osmand-bikepacking).
When I test things with OsmAndMapCreator everything works fine, but once I load it in the app I get strange effects.

## Rendering version 1
Once I move the map the previous rendering still shines through where there is no polygon defined:
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/ed916916-5539-4624-ab00-77c7be492142"  width="300" height="auto"/> </td>
  </tr>
</table>

## Rendering Version 2 (OpenGL)
No lines are displayed
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/0b4ae3e5-7fcb-4b23-8516-1aea0945e247"  width="300" height="auto"/> </td>
  </tr>
</table>

Can anyone hint me to what I'm doing wrong in `bikepacking.render.xml`?
Thanks in advance!

**Comment by vshcherb** [comment]
Your style is quite complex already to dig into quickly. Probably you can start assigning https://github.com/hhornburg/osmand-bikepacking/blob/main/bikepacking.render.xml#L632 strokeWidth constant and color instead of evaluated attributes. Also please check strokeWidth_1, color_1 - though if strokeWidth is 0 then line won't be displayed.
I didn't find any specific issue

**Reply by hhornburg** [reply]
I also noticed that the assigning of the `strokeWidth` in the `renderAttribute` (e.g. in https://github.com/hhornburg/osmand-bikepacking/blob/49e8f9b129be466f975f7c6097a5ee0519dc5133/bikepacking.render.xml#L106) stops working once I add the following snippet:
```xml
    <renderingAttribute name="shadowRendering">
        <!-- 0 - no shadow, 1 - one step, 2 - blur shadow, 3 - solid shadow -->
        <case attrIntValue="3"/>
    </renderingAttribute>
```
I assume assigning the width via side-effect was never the intended way to go, but using `attrFloatValue` in the `renderAttribute`  had no effect in OsmAndMapCreator.

Is this an issue of OsmAndMapCreator, so I have to copy over each style to the device to test it or is there a some `renderAttribute` (like `shadowRendering`) missing that changes settings for the renderer?

TBH I'm pretty puzzled due to the lack of documentation (that I found at least) on what the 'correct' way regarding the writing of a rendering style is.

**Reply by vshcherb** [reply]
I think it's correct way you do but noone knows did you encounter a bug or you made a typo somewhere instead of 1 -> 1.1  and opengl engine doesn't like float numbers somewhere or something like that. Though I believe width could be float, at least it should support syntax "4:2"

**Reply by hhornburg** [reply]
I created a repo where I tested the different combinations: https://github.com/hhornburg/osmand-rendertests/tree/main/shadowRendering

My result is: set both `attrFloatValue` and `attrIntValue` in a `renderingAttribute` to have identical output for the `strokeWidth` in the app and the map creator. Don't use Renderer Version 1 in the app.
